tests/llama_index/test_llama_index_tracer.py

```diff
@@ -180,72 +256,43 @@ def test_trace_retriever(multi_index, is_async):
     assert spans[3].attributes["model_name"] == Settings.embed_model.model_name
 
 
-@pytest.mark.parametrize("is_async", [True, False])
-def test_trace_query_engine(multi_index, is_async):
-    engine = multi_index.as_query_engine()
-
-    response = asyncio.run(engine.aquery("Hello")) if is_async else engine.query("Hello")
-    assert response.response.startswith('[{"role": "system", "content": "You are an')
-    response = asdict(response)
+@pytest.mark.parametrize(
+    ("is_stream", "is_async"),
+    [
+        (False, False),
+        (True, False),
+        (False, True),
+        # (True, True),  # Async stream is not supported yet
+    ],
+)
+def test_trace_query_engine(multi_index, is_stream, is_async):
+    engine = multi_index.as_query_engine(streaming=is_stream)
+
+    if is_stream:
+        response = engine.query("Hello")
+        assert isinstance(response, StreamingResponse)
+        assert len(_get_all_traces()) == 0
+        response = "".join(response.response_gen)
+        assert response == "Hello world"
+    else:
+        response = asyncio.run(engine.aquery("Hello")) if is_async else engine.query("Hello")
+        assert response.response.startswith('[{"role": "system", "content": "You are an')
+        response = asdict(response)
 
     traces = _get_all_traces()
     assert len(traces) == 1
     assert traces[0].info.status == TraceStatus.OK
 
-    spans = traces[0].data.spans
-    assert len(spans) == 13 if is_async else 14
-
-    # Validate the tree structure
```

Comment:
note: This check seems to be too rigorous. The underlying operations in engines is dynamic and change depending on library version.